content/en/docs/concepts/storage/persistent-volumes.md

```diff
@@ -785,6 +785,65 @@ spec:
       storage: 10Gi
 ```
 
+## Volume Populators and Data Sources
+
+{{< feature-state for_k8s_version="v1.22" state="alpha" >}}
+
+{{< note >}}
+Volume populars are available as alpha since Kubernetes 1.18, are redesigned in Kubernetes 1.22. The `AnyVolumeDataSource` feature must be enabled. Refer to the [feature gate](/docs/reference/command-line-tools-reference/feature-gates/) documentation for more information.
+{{< /note >}}
+
+Volume populators take advantage of a new PVC spec field called `dataSourceRef`. Unlike the
+`dataSource` field, which can only contain either a reference to another PersistentVolumeClaim
+or to a VolumeSnapshot, the `dataSourceRef` field can contain a reference to any object in the
+same namespace, except for core objects other than PVCs.
+
+### Data Source Field Semantics
```

Comment:
Currently under https://kubernetes.io/docs/concepts/storage/persistent-volumes/, we have "Volume Snapshot and Restore Volume from Snapshot Support" and "Volume Cloning" at the same level.  If we merge this new volumeDataSource with Volume Cloning, then we need to merge "Restore Volume from Snapshot" into the same heading as well.  Otherwise, it's better to have the 3 of them under different headings at the same level.

Reply:
The concept we want to explain is that you can define a way to claim a PersistentVolume where that PV arrives already containing data. I'd make this be a second-level heading. Having introduced that generally, we should also enumerate how you can achieve that:
- cloning a PVC that already exists
- using a snapshot as the source for populating a PV using a `VolumeSnapshot` datasource (hyperlinking to https://kubernetes.io/docs/concepts/storage/volume-snapshots/ for more details).
- using a custom populator as the source for populating a PV

I'd make each of those be third-level headings under the general heading. This is because they describe different ways to end up with a ready-populated PV and, in fact, they use the same fields in the API.

Potentially, we could move all of these concepts to a new page about populating PVs during creation. For now I'd leave them in the Persistent Volumes topic.

Reply:
I like the idea of tabs and reworking this section but I am not good at layout design. My change was focused on adding the new information that's changed since the last release. I worry that by reworking everything we might lose existing information.

Reply:
OK, that's fair enough. Let's focus on getting the content _accurate_ and then we can try to do a follow up PR that makes it more readable. Ideally we can get that follow-up PR in the dev-1.22 branch before the release, but it's not the end of the world if it goes in later.

@bswartz could you point out https://github.com/kubernetes/website/pull/28610#discussion_r677488539 in the PR description? This is to call out that we're taking a small shortcut with this PR, and to help frame how we review it.

Reply:
Updated